### PR TITLE
Use email templates to generate the mail messages

### PR DESCRIPTION
**Pull request by jvillafanez**
Use templates to generate the email message

### CODE REVIEW

File: templates/mail/htmlmail.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+</tr>
+<tr><td colspan="2">&nbsp;</td></tr>
+<tr>
+<td width="20px">&nbsp;</td>
+<td style="font-weight:normal; font-size:0.8em; line-height:1.2em; font-family:verdana,'arial',sans;">
+<?php p($_['subject']); ?>
+<br /><br />
+<?php p($_['message']); ?>
+<br /><br />
+<?php print_unescaped($l->t('Go to <a href="%s">%s</a> to check the notification', [$_['serverUrl'], $_['serverUrl']])); ?>
```
Comment (DeepDiver1975) [inline templates/mail/htmlmail.php]: each and every print_unescaped needs a close look so that no xss is possible

Comment (jvillafanez) [reply templates/mail/htmlmail.php]: The serverUrl variable is generated in https://github.com/owncloud/notifications/blob/master/lib/Mailer/NotificationMailerAdapter.php#L88 .
I guess it depends on the level of trust we have on the link provided by the notification or the one we're generating. We can harden the check to verify it's an url.

The other thing is the translation.

**Comment by jvillafanez** [comment]
Rebased

**Comment by jvillafanez** [comment]
Backport in https://github.com/owncloud/notifications/pull/162

File: templates/mail/htmlmail.php (hunk 1 above)
Comment (settermjd) [reply templates/mail/htmlmail.php]: I'm a bit late to this, but shouldn't the templates always auto-escape template variables?

Comment (PVince81) [reply templates/mail/htmlmail.php]: depends if it's escaped earlier or not. you don't want to double escape as it can make strings look odd

**Pull request merged**